backends/dpdk/dpdkAsmOpt.cpp

```diff
@@ -371,6 +377,119 @@ CopyPropagationAndElimination::copyPropAndDeadCodeElim(
+                    replaceIfCopy(eq->src1), replaceIfCopy(eq->src2)));
+       } else if (auto xor1 = stmt->to<IR::DpdkXorStatement>()) {
+            instr.push_back(new IR::DpdkXorStatement(replaceIfCopy(xor1->dst, false),
+                    replaceIfCopy(xor1->src1), replaceIfCopy(xor1->src2)));
+       } else if (auto cmp = stmt->to<IR::DpdkCmpStatement>()) {
+            instr.push_back(new IR::DpdkCmpStatement(replaceIfCopy(cmp->dst, false),
+                    replaceIfCopy(cmp->src1), replaceIfCopy(cmp->src2)));
+       } else if (auto and1 = stmt->to<IR::DpdkLAndStatement>()) {
+            instr.push_back(new IR::DpdkLAndStatement(replaceIfCopy(and1->dst, false),
+                    replaceIfCopy(and1->src1), replaceIfCopy(and1->src2)));
+       } else if (auto lor = stmt->to<IR::DpdkLOrStatement>()) {
+            instr.push_back(new IR::DpdkLOrStatement(replaceIfCopy(lor->dst, false),
+                    replaceIfCopy(lor->src1), replaceIfCopy(lor->src2)));
+       } else if (auto leq = stmt->to<IR::DpdkLeqStatement>()) {
+            instr.push_back(new IR::DpdkLeqStatement(replaceIfCopy(leq->dst, false),
+                    replaceIfCopy(leq->src1), replaceIfCopy(leq->src2)));
+       } else if (auto lss = stmt->to<IR::DpdkLssStatement>()) {
+            instr.push_back(new IR::DpdkLssStatement(replaceIfCopy(lss->dst, false),
+                    replaceIfCopy(lss->src1), replaceIfCopy(lss->src2)));
+       } else if (auto grt = stmt->to<IR::DpdkGrtStatement>()) {
+            instr.push_back(new IR::DpdkGrtStatement(replaceIfCopy(grt->dst, false),
+                    replaceIfCopy(grt->src1), replaceIfCopy(grt->src2)));
+       } else if (auto geq = stmt->to<IR::DpdkGeqStatement>()) {
+            instr.push_back(new IR::DpdkGeqStatement(replaceIfCopy(geq->dst, false),
+                    replaceIfCopy(geq->src1), replaceIfCopy(geq->src2)));
+       } else if (auto neq = stmt->to<IR::DpdkNeqStatement>()) {
+            instr.push_back(new IR::DpdkNeqStatement(replaceIfCopy(neq->dst, false),
+                    replaceIfCopy(neq->src1), replaceIfCopy(neq->src2)));
+       } else if (auto recd = stmt->to<IR::DpdkRecircidStatement>()) {
+            instr.push_back(new IR::DpdkRecircidStatement(replaceIfCopy(recd->pass, false)));
+       } else if (auto rarm = stmt->to<IR::DpdkRearmStatement>()) {
+            instr.push_back(new IR::DpdkRearmStatement(replaceIfCopy(rarm->timeout)));
+       } else if (auto csum = stmt->to<IR::DpdkChecksumAddStatement>()) {
+            instr.push_back(new IR::DpdkChecksumAddStatement(csum->csum, csum->intermediate_value,
+                    replaceIfCopy(csum->field)));
+       } else if (auto csum = stmt->to<IR::DpdkChecksumSubStatement>()) {
+            instr.push_back(new IR::DpdkChecksumSubStatement(csum->csum, csum->intermediate_value,
+                    replaceIfCopy(csum->field)));
+       } else if (auto hash = stmt->to<IR::DpdkGetHashStatement>()) {
```

Comment:
Lets not remove moves inserted for Hash. Hash parameters are moved to metadata if they are not consecutive.

Reply:
ok